File: csharp/hamming/Hamming.cs
using System;

public static class Hamming
{
    public static int Distance(string firstStrand, string secondStrand)
    {
        var length = firstStrand.Length;

        if (length != secondStrand.Length)
            throw new ArgumentException();

        var distance = 0;

        for (int i = 0; i < length; i++)
        {
            if (firstStrand[i] != secondStrand[i])
                distance++;
        }

        return distance;
    }
}

File: go/hamming/hamming.go
package hamming

import "errors"

func Distance(a, b string) (distance int, err error) {
	aRunes := []rune(a)
	bRunes := []rune(b)

	if len(aRunes) != len(bRunes) {
		return distance, errors.New("Strings must be of equal length")
	}

	for i, r := range aRunes {
		if r != bRunes[i] {
			distance++
		}
	}

	return
}
